Makefile

```diff
@@ -451,6 +451,43 @@ st-checks:
 	# running on the host.
 	iptables-save | grep -q 'calico-st-allow-etcd' || iptables $(IPT_ALLOW_ETCD)
 
+.PHONY: k8s-test
+## Run the k8s tests
+k8s-test: $(NODE_CONTAINER_CREATED) calico_test.created
+	wget https://raw.githubusercontent.com/lwr20/kubeadm-dind-cluster/master/fixed/dind-cluster-v1.10.sh
+	chmod +x dind-cluster-v1.10.sh
+	$(MAKE) k8s-stop
+	$(MAKE) k8s-start
+	docker run \
+	    -v $(CURDIR):/code \
+	    -v /var/run/docker.sock:/var/run/docker.sock \
+	    -v /home/$(USER)/.kube/config:/root/.kube/config \
+	    -v /home/$(USER)/.kubeadm-dind-cluster:/root/.kubeadm-dind-cluster \
+	    --privileged \
+	    --net host \
+        $(TEST_CONTAINER_NAME) \
+	    sh -c 'cp /root/.kubeadm-dind-cluster/kubectl /bin/kubectl && ls -ltr /bin/kubectl && which kubectl && cd /code/tests/k8st && nosetests -v --with-xunit --xunit-file="/code/report/k8s-tests.xml" --with-timer'
+	$(MAKE) k8s-stop
+
+.PHONY: k8s-start
+## Start k8s cluster
+k8s-start:
+	CNI_PLUGIN=calico ./dind-cluster-v1.10.sh up
+	docker save $(BUILD_IMAGE):latest-$(ARCH) > calico-node.tar
+	docker cp calico-node.tar kube-master:/calico-node.tar
+	docker cp calico-node.tar kube-node-1:/calico-node.tar
+	docker cp calico-node.tar kube-node-2:/calico-node.tar
+	docker exec kube-master docker load -i /calico-node.tar
+	docker exec kube-node-1 docker load -i /calico-node.tar
+	docker exec kube-node-2 docker load -i /calico-node.tar
+	kubectl apply -f  https://docs.projectcalico.org/v3.3/getting-started/kubernetes/installation/hosted/calicoctl.yaml
```

Comment:
It's calicoctl so that its available to the tests to use.  One of the recommended ways to install calicoctl these days is using that manifest.  Then when you want to run a calicoctl command, you just `kubectl exec` in that pod.  

The manifest picks up the data store config from the cluster itself.